### ISSUE TITLE
CMake common commands wrap

### ISSUE THREAD

**Issue by memsharded** [conan-io/conan#759]
Suggested by @Manu343726, simplify some common ``CMake`` commands:

```python
self.run("cmake %s ..." % self.conanfile_directory) => cmake.configure(self)
self.run("cmake --build . %s " % ....) => cmake.build(self)
```

Related to https://github.com/conan-io/conan/issues/755, as this would alleviate the problem with path with spaces while invoking cmake

**Comment by lasote** [comment]
Maybe opening the configure/build commands to accept a list of additional parameters? It's very common to add a definition, and also prepend some environment command, so, maybe `self.run(cmake.configure_command)`?

**Comment by memsharded** [comment]
@Manu343726 you have now available your cmake wrapped calls, courtesy of @sixten-hilborn 
Released in 0.20, to be documented (https://github.com/conan-io/docs/issues/158), but checking the PR, it should be easy to figure out.
